From akpm@osdl.org Tue Oct 10 14:51:06 2006
Message-Id: <200610102142.k9ALgolP025527@shell0.pdx.osdl.net>
From: Eric Sesterhenn <snakebyte@gmx.de>
Subject: USB: Memory leak in drivers/usb/serial/airprime.c
To: greg@kroah.com
Cc: akpm@osdl.org, snakebyte@gmx.de, andy@andynet.net
Date: Tue, 10 Oct 2006 14:42:50 -0700

From: Eric Sesterhenn <snakebyte@gmx.de>

the commit
http://www.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=5dda171202f94127e49c12daf780cdae1b4e668b
added a memory leak.  In case we cant allocate an urb, we dont free the
buffer and leak it.  Coverity id #1438

Signed-off-by: Eric Sesterhenn <snakebyte@gmx.de>
Acked-by: Andy Gay <andy@andynet.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/airprime.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/usb/serial/airprime.c
+++ gregkh-2.6/drivers/usb/serial/airprime.c
@@ -134,6 +134,7 @@ static int airprime_open(struct usb_seri
 		}
 		urb = usb_alloc_urb(0, GFP_KERNEL);
 		if (!urb) {
+			kfree(buffer);
 			dev_err(&port->dev, "%s - no more urbs?\n",
 				__FUNCTION__);
 			result = -ENOMEM;
